src/generate-types.ts

```diff
@@ -0,0 +1,98 @@
+// @ts-nocheck
+module.exports = {
+  /**
+   * Generates TypeScript types based on the provided GraphQL schema.
+   * @param schema - The GraphQL schema object to generate types from.
+   * @param _documents - Unused parameter for additional documents.
+   * @param _config - Unused parameter for configuration options.
+   * @returns A string containing the generated TypeScript types.
+   */
+  plugin: function (schema, _documents, _config) {
+    if (!schema.getQueryType()) {
+      throw new Error('No query type found in schema')
+    }
+
+    /**
+     * Capitalizes the first letter of the given string.
+     * @param str - The string to capitalize.
+     * @returns The input string with the first letter converted to uppercase.
+     */
+    function capitalize (str) {
+      return str.charAt(0).toUpperCase() + str.slice(1)
+    }
+
+    /**
+     * Generates TypeScript types for preload operations.
+     * @returns A string containing the TypeScript type definitions for preload operations.
+     */
+    function printPreloadTypes () {
+      const extractFieldsType =
```

Comment:
This type we can promote to our `types.ts` file as it's not used in the generated types